modules/installing-ocp-agent.adoc

```diff
@@ -0,0 +1,171 @@
+// Module included in the following assemblies:
+//
+// * installing-with-agent/installing-with-agent.adoc
+
+:_content-type: PROCEDURE
+[id="installing-ocp-agent_{context}"]
+= Installing {product-title} with Agent
+
+You can use the Agent-based installer to install a:
+
+* Single-node cluster
+* Compact cluster
+* Highly Available (HA) cluster
+
+The following procedure deploys a single-node {product-title} and you can use this procedure as a basis and modify according to your requirements.
+
+.Prerequisites
+* RHEL 8.6
+* 32GB memory
+* 120GB disk
+* 8 CPUs
+
+.Procedure
+
+. Navigate to the link:https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/latest-{product-version}/[OpenShift mirror site] and download the latest version of the tarball that matches your operating system.
+
+. Extract the binary.
++
+[source,terminal]
+----
+$ tar xvf <file> -C ~/.local/bin
+----
++
+The file contains the Agent-based installer.
+
+. Install `nmstate` dependency.
++
+[source,terminal]
+----
+$ sudo dnf install /usr/bin/nmstatectl -y
+----
+
+. Place the `openshift-install` binary in a directory that is on your PATH.
+
+. Create a directory to store the cluster manifests.
++
+[source,terminal]
+----
+$ mkdir ~/<directory_name>
+----
+
+. Create the `install-config.yaml` file.
++
+[source,yaml]
+----
+  cat << EOF > ./cluster-manifests/install-config.yaml
+  apiVersion: v1
+baseDomain: test.example.com
+compute:
+- architecture: amd64
+  hyperthreading: Enabled
+  name: worker
+  replicas: 0
+controlPlane:
+  architecture: amd64
+  hyperthreading: Enabled
+  name: master
+  replicas: 1
+metadata:
+  name: sno-cluster <1>
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineNetwork:
+  - cidr: 192.168.111.0/16
+  networkType: OpenShiftSDN
+  serviceNetwork:
+  - 172.30.0.0/16
+platform:
+  none: {}
+pullSecret: '<pull_secret>' <2>
+sshKey: |
+  '<ssh_pub_key>' <3>
```

Comment:
Hi @Amrita42, I think we should take a reference of [baremetal Installation docs](https://docs.openshift.com/container-platform/4.11/installing/installing_bare_metal/installing-bare-metal.html#installation-bare-metal-config-yaml_installing-bare-metal) to explain each value of the `install-config` and also explain the value of these fields specifically for each topology `compute. replicas`, `controlPlane.replicas` and `platform`. Machine network (networking.machineNetwork) explanation can be found [here](https://docs.openshift.com/container-platform/4.11/installing/installing_bare_metal/installing-bare-metal-network-customizations.html#installation-configuration-parameters-network_installing-bare-metal-network-customizations).

Reply:
When creating an SNO cluster ( i.e. platform is none) then the `networking.networkType` has to be `OVNKubernetes` otherwise `OpenShiftSDN`

Reply:
Also, the CIDRs and the ServiceNetwork IP values can be different

Reply:
there is related to  jira https://issues.redhat.com/browse/AGENT-189 , I will document this in the next PR.